### PR TITLE
 feat: Add derive macro for PrefabData

### PR DESCRIPTION
**Pull request by Rhuagh**
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/1035)

### CODE REVIEW

**Comment by Rhuagh** [comment]
Left to do: derive most of the internal implementations instead of custom impls.

**Comment by jojolepro** [comment]
Hey, that would be pretty useful in my ui rework :)

**Comment by Moxinilian** [comment]
This is really beautiful. Thank you so much!

File: book/src/prefab.md [hunk 1]
```diff
@@ -198,6 +198,87 @@ There are a few special blanket implementations provided by the asset system:
...
+pub struct SomeComponent {
+    pub id: u64,
+}
+
+impl Component for SomeComponent {
+    type Storage = DenseVecStorage<Self>; 
+}
+```
+
+You might notice here that `SomeComponent` have no `PrefabData` derive on its own, it is simply
```
Comment (LucioFranco) [inline book/src/prefab.md]: `have` -> `has`

**Comment by Rhuagh** [comment]
bors r=torkleyy,jojolepro,LucioFranco

**Pull request merged**